cs41_final.sh

```diff
@@ -74,59 +176,77 @@ while [ userChoice != '0' ]
       case $userChoice in
         1) clear;
         option_picked "Option 1: List of current Users";
-        #do something
+        currentUsers
         show_menu;
         ;;
 
         2) clear;
         option_picked "Option 2: List of active processes for user ${userName}";
-        #do something
-        show_menu;
+        echo "Currently logged-in users:";
+        currentUsers;
+        echo "Please select from available list.";
+        read userName;
+        #do something: check for validity, if not valid, prompt again. when true do something
+        listActiveProcess;
+        show_menu;       show_menu;
         ;;
 
         3) clear;
-        option_picked "Option 3: Active processes for user ${userName}";
-        #do something;
+        option_picked "Option 3: Monitor active processes for user ${userName}";
+        echo "Currently logged-in users:";
+        currentUsers;
+        echo "Please select from available list.";
+        read userName;
+        #do something: check for validity, if not valid, prompt again. when true do something
+        monitorActiveProcess;
         show_menu;
         ;;
 
         4) clear;
         option_picked "Option 4: Kill Process";
-        #do something
+        killProcess;
         show_menu;
         ;;
 
         5) clear
         option_picked "Option 5: Show last 10 users";
+        lastUsers; 
         #do something
         show_menu;
         ;;
 
         6) clear;
         option_picked "Option 6: Show free disk space";
-        #do something
+        freeDiskSpace;
         show_menu;
         ;;
 
         7) clear;
         option_picked "Option 7: Show disk space for user home directory";
-        #do something
+        echo "Currently logged-in users:";
+        currentUsers;
+        echo "Please select from available list.";
+        read userName;
+        #do something: check for validity, if not valid, prompt again. when true do something
+        userDiskSpace;
         show_menu;
         ;;
         
         8) clear;
         option_picked "Option 8: Find files larger than 100MB";
-        #do something
+        fileGE100MB;
         show_menu;
         ;;
 
         9) clear;
         option_picked "Option 9: Live Monitor";
-        #do something
+        liveMonitor;
         show_menu;
         ;;
 
-        0)exit;
+        0) clear;
+        goodBye;
+        exit;
         ;;
 
       esac
```

Comment:
You're off to a good start.  Now time to start fleshing out each option's details, verifying they all work!